### PR TITLE
[EPICSYSTEM-11] add project search and results table components

### PR DESCRIPTION
**Pull request by Ckoelewyn**

### CODE REVIEW

**Comment by rpyyj** [comment]
Please resolve the conflicts

**Comment by rpyyj** [comment]
Whoa! It's great to see actual search results! Apologies for taking so long with the review but there is a lot to unpack.

File: src/components/Filter.js [hunk 1]
```diff
@@ -11,6 +11,9 @@ import Checkbox from "@mui/material/Checkbox";
 import ExpandLessIcon from "@mui/icons-material/ExpandLess";
 import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
 
+import { PROJECT_LEGISLATION_YEARS } from "constants";
```
Comment (rpyyj) [inline src/components/Filter.js]: Please remove these unused constant imports. (These show up as `eslint` warnings in your `npm start` console.)

File: src/components/Filter.js [hunk 2]
```diff
@@ -56,11 +69,10 @@ const useStyles = makeStyles()((theme, { width }) => ({
 	},
 }));
 
-const Filter = ({ icon, items = [], onChange, selected, title }) => {
+const Filter = ({ filterKey, icon, items = [], onChange, selected, title }) => {
```
Comment (rpyyj) [inline src/components/Filter.js]: Please remove the `filterKey` prop

File: src/components/Filter.js [hunk 3]
```diff
@@ -131,6 +144,7 @@ const Filter = ({ icon, items = [], onChange, selected, title }) => {
 };
 
 Filter.propTypes = {
+	filterKey: PropTypes.string,
```
Comment (rpyyj) [inline src/components/Filter.js]: Please remove the `filterKey` prop type

File: src/components/Filter.js [hunk 4]
```diff
@@ -56,11 +69,10 @@ const useStyles = makeStyles()((theme, { width }) => ({
 	},
 }));
 
-const Filter = ({ icon, items = [], onChange, selected, title }) => {
+const Filter = ({ filterKey, icon, items = [], onChange, selected, title }) => {
 	const [keys, setKeys] = useState(selected.map(({ key }) => key));
 	const [open, setOpen] = useState(false);
 	const [width, setWidth] = useState(0);
-
```
Comment (rpyyj) [inline src/components/Filter.js]: Please restore this line break between the state definitions and the hook call

File: src/components/Filter.js [hunk 5]
```diff
@@ -38,16 +43,24 @@ const useStyles = makeStyles()((theme, { width }) => ({
...
 		boxShadow: "2px 6px 8px 0px #0000001A",
+		border: "1px solid #EEEEEE",
+		borderRadius: "0 0 0.25rem 0.25rem",
 		width: width,
 		"& ul": {
 			listStyleType: "none",
 			margin: 0,
 			padding: 0,
 		},
+		overflowY: "auto",
```
Comment (rpyyj) [inline src/components/Filter.js]: Thank you for stopping the never-ending menu madness! Please move these two lines above `width` to keep the 🔤

File: src/components/Results/ResultsPagination/PagePreviousButton/index.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+		handleChangePage(event, pageNum - 1);
+	};
+
+	useEffect(() => {
+		setIsDisabled(pageNum === 0);
+	}, [pageNum, pageSize, setIsDisabled, totalResultCount]);
+
+	return (
+		<div className={classes.container}>
+			<IconButton aria-label={label} disabled={isDisabled} onClick={handleOnClick} type="button">
```
Comment (rpyyj) [inline src/components/Results/ResultsPagination/PagePreviousButton/index.js]: Reminder to use `<Button>` here

**Review by rpyyj — CHANGES_REQUESTED** [review]

File: src/components/Filter.js (hunk 1 above)
Comment (Ckoelewyn) [reply src/components/Filter.js]: Updated in [b425df8](https://github.com/bcgov/epic-public/pull/3/commits/b425df8adcb5800bca27d62866f23b734f4aa0cc)

File: src/components/Filter.js (hunk 2 above)
Comment (Ckoelewyn) [reply src/components/Filter.js]: updated in [b425df8](https://github.com/bcgov/epic-public/pull/3/commits/b425df8adcb5800bca27d62866f23b734f4aa0cc)

File: src/components/Filter.js (hunk 4 above)
Comment (Ckoelewyn) [reply src/components/Filter.js]: updated in [b425df8](https://github.com/bcgov/epic-public/pull/3/commits/b425df8adcb5800bca27d62866f23b734f4aa0cc)

File: src/components/Results/ResultsPagination/PagePreviousButton/index.js (hunk 6 above)
Comment (Ckoelewyn) [reply src/components/Results/ResultsPagination/PagePreviousButton/index.js]: updated in [b425df8](https://github.com/bcgov/epic-public/pull/3/commits/b425df8adcb5800bca27d62866f23b734f4aa0cc)

File: src/components/Filter.js (hunk 5 above)
Comment (Ckoelewyn) [reply src/components/Filter.js]: updated in [b425df8](https://github.com/bcgov/epic-public/pull/3/commits/b425df8adcb5800bca27d62866f23b734f4aa0cc)

File: src/components/Filter.js (hunk 3 above)
Comment (Ckoelewyn) [reply src/components/Filter.js]: updated in [b425df8](https://github.com/bcgov/epic-public/pull/3/commits/b425df8adcb5800bca27d62866f23b734f4aa0cc)

**Review by marklise — APPROVED** [review]

**Pull request merged**